docs/API/LorisRESTAPI.md

```diff
@@ -0,0 +1,460 @@
+        },
+        "Visit" : {
+            "Date" : "YYYY-MM-DD",
+            "Status" : "Pass|Failure|Withdrawal|In Progress"
+        },
+        "Approval" : {
+            "Date" : "YYYY-MM-DD",
+            "Status" : "Pass|Failure|Withdrawal|In Progress"
+        }
+    }
+}
+```
+
+A PUT of the same format but with only the Meta fields will create the VisitLabel
+for this candidate, in an unstarted stage if the Visit label provided is valid.
+
+PATCH is not supported for Visit Labels.
+
+It will return a 404 Not Found if the visit label does not exist for this candidate
+(as well as anything under the /candidates/$CandID/$VisitLabel hierarchy)
+
+Any of the Stages may not be present in the returned result if the stage has not
+started yet or is not enabled for this project (ie. if useScreening is false in
+Loris, or Approval has not occured)
+
+### 3.3 Candidate Instruments
+```
+GET /candidates/$CandID/$VisitLabel/instruments
+```
+
+Will return a JSON object of the form.
+
+```js
+{
+    "Meta" : {
+        "CandID" : CandID,
+        "Visit"  : VisitLabel
+    },
+    "Instruments" : [ "InstrumentName", "AnotherInstrument", ...]
+}
+```
+
+Where the instruments array represents the instruments that were administered for that
+candidate at that visit. InstrumentNames are the short names and the forms for them
+SHOULD all be retrievable through the `project` portion of the API.
+
+PUT / PATCH / POST are not currently supported for candidate instruments.
+
+### 3.3 The Candidate Instrument Data
+
+```
+GET /candidates/$CandID/$VisitLabel/instruments/$InstrumentName[/dde]
+PUT /candidates/$CandID/$VisitLabel/instruments/$InstrumentName[/dde]
+PATCH /candidates/$CandID/$VisitLabel/instruments/$InstrumentName[/dde]
+```
+
+These will retrieve or modifiy the data for $InstrumentName. If /dde is present, the double data
+entry form of the data will be retrieved/modified. If absent, the "single data entry" version
+of the form is used instead.
+
+The format returned by a GET request is a JSON document of the form:
+
+```js
+{
+    "$InstrumentName" : {
+        "FieldName1" : "Value1",
+        "FieldName2" : "Value2",
+        ...
+    }
+}
+```
+
+Including the values of ALL fields for this instrument, including score field values.
+
+The body of a PUT request to the same URL MUST contain a JSON object of the same format. Data PUT
+to the URL SHOULD contain all fields with data entry. The server will null the data for keys not
+specified. A PUT request MAY not specify score columns that will be calculated/overwriten by
+server-side scoring. If the client attempted to score fields client-side and the value passed by the PUT
+request conflict with the server-side calculation of the score, the server-side calculation will win.
+Any keys specified in the document PUT that do not match a corresponding field in the form MAY be ignored.
+
+The specification for PATCH request is similar to a PUT request, with the exception that any
```

Comment:
Patch doesn't seem to handled by the API. There is no case in the ```handleRequest()``` in  ```APIBases.php``` for PATCH and there is no ```handlePatch()``` method in ```InstrumentData.php```